src/car/v1/mod.rs

```diff
@@ -40,14 +40,15 @@ impl CarV1 {
         // End of the header
         let header_end = r.stream_position()?;
         // If we're in a CARv2
-        if let Some(index_offset) = index_offset
-            && r.seek(SeekFrom::Start(index_offset)).is_ok()
-            && let Ok(index) = <Index<Bucket>>::read_bytes(&mut r)
-        {
-            return Ok(Self {
-                header,
-                index: RefCell::new(index),
-            });
+        if let Some(index_offset) = index_offset {
+            if r.seek(SeekFrom::Start(index_offset)).is_ok() {
+                if let Ok(index) = <Index<Bucket>>::read_bytes(&mut r) {
+                    return Ok(Self {
+                        header,
+                        index: RefCell::new(index),
+                    });
+                }
+            }
```

Comment:
This has the same problem as the disk store, there are errors here that should probably be handled. The documentation claims the car file should already be seeked to the correct location so I'm not sure why the offset is being provided here or any seeking is occurring...